### ISSUE TITLE
Command line autocompletion is broken for shell commands

### ISSUE THREAD

**Issue by andrey-starodubtsev** [vim/vim#1435]
vim is built from 7a073549

<Ctrl-D> prints autocomplete options twice:
```
:e ~/An<Ctrl-D>
Android/  Anki/
:!~/An<Ctrl-D>
Android/  Anki/     Android/  Anki/
```

<Ctrl-L> doesn't add next path component if there's the only option:
```
:e ~/And<Ctrl+L>
:e ~/Android/<Ctrl+L>
:e ~/Android/Sdk/
:!~/And<Ctrl+L>
:!~/Android/<Ctrl+L>
:!~/Android/
```

**Comment by chrisbra** [comment]
partly duplicate of #539, patch with test [posted here](https://groups.google.com/d/msg/vim_dev/Cwf7xZAqo9o/3yX1eU8kCgAJ)

**Comment by lacygoill** [comment]
Both issues are fixed by [8.1.0017](https://github.com/vim/vim/releases/tag/v8.1.0017).

**Comment by chrisbra** [comment]
thanks, closing.
